### PR TITLE
fix(selenium,playwright): only set `allowedOrigin` when needed

### PR DESCRIPTION
**Pull request by Zidious**

### CODE REVIEW

File: playwright/package.json [hunk 1]
```diff
@@ -5,12 +5,13 @@
   "private": true,
   "repository": "dequelabs/axe-core-maven-html",
   "scripts": {
-    "start": "http-server ./src/test/resources/fixtures -p 1337"
+    "start": "http-server ./node_modules/axe-test-fixtures/fixtures -p 1337"
```
Comment (Zidious) [inline playwright/package.json:8]: I cannot remember why I put all of the text-fixtures in the playwright `resources/fixtures` that was not intended. Now we pull from `node_modules` instead.

File: playwright/src/test/java/PlaywrightJavaTest.java [hunk 2]
```diff
@@ -548,50 +546,6 @@ public void withMultipleIncludes() {
     assertEquals(axeResults.getPasses().get(0).getNodes().size(), 2);
   }
 
-  @Test
-  public void withIncludeAndExclude() {
```
Comment (Zidious) [inline playwright/src/test/java/PlaywrightJavaTest.java:552]: removed as not in the current fixtures

File: playwright/src/test/java/PlaywrightJavaTest.java [hunk 3]
```diff
@@ -548,50 +546,6 @@ public void withMultipleIncludes() {
...
-
-    assertTrue(targets.stream().anyMatch(selector -> selector.equalsIgnoreCase(".include-nested")));
-    assertTrue(
-        targets.stream().anyMatch(selector -> selector.equalsIgnoreCase(".include-nested2")));
-    assertTrue(
-        targets.stream().noneMatch(selector -> selector.equalsIgnoreCase(".exclude-nested")));
-  }
-
-  @Test
-  public void withIncludeAndExcludeIframes() {
```
Comment (Zidious) [inline playwright/src/test/java/PlaywrightJavaTest.java:572]: removed as not in the current fixtures

**Review by michael-siek — APPROVED** [review]

File: playwright/src/test/java/AllowedOriginsTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    browser.close();
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndNotLegacyMode() {
+    page.navigate(addr() + "/index.html");
+    new AxeBuilder(page).analyze();
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```
Comment (straker) [inline playwright/src/test/java/AllowedOriginsTest.java:102]: Will need to import `java.util.Objects`

```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: playwright/src/test/java/AllowedOriginsTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndLegacyMode() {
+    page.navigate(addr() + "/index.html");
+
+    new AxeBuilder(page).setLegacyMode(true).analyze();
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```
Comment (straker) [inline playwright/src/test/java/AllowedOriginsTest.java:112]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: playwright/src/test/java/AllowedOriginsTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+  @Test
+  public void shouldNotSetWhenRunningLegacySourceAndLegacyMode() throws Exception {
+    page.navigate(addr() + "/index.html");
+    overwriteAxeSourceWithString(downloadFromURL(addr() + "/axe-core@legacy.js") + axeForceLegacy);
+
+    new AxeBuilder(page).setLegacyMode(true).analyze();
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```
Comment (straker) [inline playwright/src/test/java/AllowedOriginsTest.java:123]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: playwright/src/test/java/AllowedOriginsTest.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+  @Test
+  public void shouldSetWhenRunningLegacySourceAndNotLegacyMode() throws Exception {
+    page.navigate(addr() + "/index.html");
+    overwriteAxeSourceWithString(downloadFromURL(addr() + "/axe-core@legacy.js") + axeForceLegacy);
+
+    new AxeBuilder(page).analyze();
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), "*");
```
Comment (straker) [inline playwright/src/test/java/AllowedOriginsTest.java:134]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add("*");
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: selenium/package.json [hunk 8]
```diff
@@ -6,7 +6,9 @@
   "repository": "dequelabs/axe-core-maven-html",
   "scripts": {},
   "dependencies": {
-    "axe-core": "^4.4.2",
-    "axe-test-fixtures": "github:dequelabs/axe-test-fixtures#v1"
+    "axe-core": "^4.4.2"
+  },
+  "devDependencies": {
+  "axe-test-fixtures": "github:dequelabs/axe-test-fixtures"
```
Comment (straker) [inline selenium/package.json:12]:
```suggestion
    "axe-test-fixtures": "github:dequelabs/axe-test-fixtures"
```

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    webDriver.quit();
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndNotLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder().analyze(webDriver);
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:59]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder().setLegacyMode(true).analyze(webDriver);
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:68]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningLegacySourceAndLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder()
+        .setLegacyMode(true)
+        .setAxeScriptProvider(new StringAxeScriptProvider(axeSource + axeForceLegacy))
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:76]: Should use legacy source (and then you don't need the force legacy script appended)

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+        .analyze(webDriver);
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:80]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+  @Test
+  public void shouldSetWhenRunningLegacySourceAndNoLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder()
+        .setAxeScriptProvider(new StringAxeScriptProvider(axeSource + axeForceLegacy))
+        .analyze(webDriver);
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), "*");
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:91]:
```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add("*");
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
+  }
+
+  @Test
+  public void shouldSetWhenRunningLegacySourceAndNoLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder()
+        .setAxeScriptProvider(new StringAxeScriptProvider(axeSource + axeForceLegacy))
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:87]: Should use legacy source (and then you don't need the force legacy script appended)

**Review by straker — DISMISSED** [review]

File: selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java [hunk 15]
```diff
@@ -0,0 +1,93 @@
+package com.deque.html.axecore.selenium;
+
+import static org.junit.Assert.*;
+
+import com.deque.html.axecore.providers.EmbeddedResourceAxeProvider;
+import com.deque.html.axecore.providers.StringAxeScriptProvider;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Scanner;
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java:10]:
```suggestion
import java.util.Scanner;
import java.util.Objects;
```

**Review by michael-siek — APPROVED** [review]

**Pull request merged**